azurelinuxagent/ga/monitor.py

```diff
@@ -329,10 +330,26 @@ def add_sysinfo(self, event):
         # Add sys_info params populated by the agent
         final_parameters.extend(self.sysinfo)
 
-        # Container id is populated for agent events on the fly, but not for extension events. Add it if it's missing.
-        if "ContainerId" not in [param.name for param in event.parameters]:
+        if "ContainerId" not in event:
+            # ContainerId is populated for agent events on the fly, but not for extension events. Add it if it's missing.
             final_parameters.append(
-                TelemetryEventParam("ContainerId", os.environ.get(CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED")))
+                TelemetryEventParam("ContainerId", get_container_id_from_env()))
+        if "GAVersion" not in event:
+            # GAVersion is populated for agent events on the fly, but not for extension events. Add it if it's missing.
+            final_parameters.append(TelemetryEventParam("GAVersion", CURRENT_AGENT))
+
+        # Default fields are only populated by Agent and not the extension. Agent will fill up any event if they don't
+        # have the default params.
+        if "OpcodeName" not in event:
+            final_parameters.append(TelemetryEventParam("OpcodeName", datetime.datetime.utcnow().__str__()))
```

Comment:
according to _add_default_parameters_to_event, the Opcode is  of event generation.  Using now() here would be incorrect. We should not populate it, or let extension owners populated in the event file to make it accurate.

Reply:
Okay. Will set it to empty.